File: January 2021/26.01.2021/Kaan.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int getDecimalValue(ListNode* head) {
        vector<int> vector;
        while(head != NULL){
            vector.push_back(head->val);
            head = head->next; 
        }
        int result = 0;
        for(int i = 0 ; i<vector.size() ; i++){
            result += vector[i] * pow(2,vector.size()-1-i);
        }
        return result;
    }
};

File: March 2021/04.03.2021/Kaan.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        
        ListNode pointerA = headA;
        ListNode pointerB = headB;
        
        int aLen = 0;
        int bLen = 0;
        

        
        while(pointerA != null){
            aLen += 1;
            pointerA = pointerA.next;
        }
        
        while(pointerB != null){
            bLen += 1;
            pointerB = pointerB.next;
        }
        
        if(aLen > bLen){
            
            int diffrence = aLen-bLen;
            
            for(int i = 0 ; i<diffrence ; i++){
                headA = headA.next;
            }
        }
        else{
            int diffrence = bLen -aLen;
            for(int i = 0 ; i<diffrence;i++){
                headB = headB.next;
            }
        }
        
        while(headA != headB ){
            headA = headA.next;
            headB = headB.next;
        }
        
        return headA;
        
    }
}
